core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -742,6 +763,14 @@ object OptionT extends OptionTInstances {
   def whenF[F[_], A](cond: Boolean)(fa: => F[A])(implicit F: Applicative[F]): OptionT[F, A] =
     if (cond) OptionT.liftF(fa) else OptionT.none[F, A]
 
+  /**
+   * Creates a non-empty `OptionT[F, A]` from an `F[A]` value if the given F-condition is considered `true`.
+   * Otherwise, `none[F, A]` is returned. Analogous to `Option.when` but for effectful conditions.
+   */
+  def whenM[F[_], A](cond: F[Boolean])(fa: => F[A])(implicit F: Monad[F]): OptionT[F, A] = OptionT(
+    F.ifM(cond)(ifTrue = F.map(fa)(Some(_)), ifFalse = F.pure(None))
+  )
```

Comment:
Just to note: although there's nothing particularly wrong with this method, however _I think_ it might not be as useful as it might seem at first glance. Just because it does not allow monads chaining. I.e. if a chain of `F[_]` ends with `F[Boolean]` then the entire chain has to be enwrapped into `whenF` to lift the result into `OptionT`.

I think, a better approach could be adding an extension method for `F[Boolean]` itself. And there's a good place for it already:
[mouse / fboolean.scala](https://github.com/typelevel/mouse/blob/main/shared/src/main/scala/mouse/fboolean.scala). E.g. it could be a new extension method of type
```
F[Boolean] => F[A] => F[Option[A]]
```
then there's another method `liftOptionT` in there https://github.com/typelevel/mouse/blob/4f995d212d9bdd375848764b89e75842463031c3/shared/src/main/scala/mouse/foption.scala#L148-L149
that can also be applied aferwards.

Reply:
we already have when and whenF. In my personal opinion, I think it is fine to add it here.

Reply:
Yep, no objections either.